### PR TITLE
Use adb wait-for-device for boot status check

### PR DESCRIPTION
**Pull request by godrei**
### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *MAJOR/MINOR/PATCH* [version update](https://semver.org/)

### Context

- Use `wait-for-device` adb command for emulator boot status check

```
adb help

 wait-for[-TRANSPORT]-STATE...
     wait for device to be in a given state
     STATE: device, recovery, rescue, sideload, bootloader, or disconnect
     TRANSPORT: usb, local, or any [default=any]
```

`adb wait-for-device` resolves issues like: `device offline`, as it only executes the given command if the emulator is in the desired `device` state.

- Updated boot status check command timeout from 20s to 1m.
- Only using `getprop sys.boot_completed` for boot complete check.
- Improve emulator unlock
- Ensure adb server is running prior to other adb calls

### Changes

- update `go-utils`, `go-steputils` and `go-android` dependencies

### CODE REVIEW

**Comment by ofalvai** [comment]
@godrei if it's not too much hassle, can you please update go-steputils in this PR? 🥺 I want [this feature](https://github.com/bitrise-io/go-steputils/pull/42) to land in as many steps as possible.

**Comment by godrei** [comment]
@ofalvai updated in https://github.com/bitrise-steplib/steps-wait-for-android-emulator/pull/25/commits/b40696f124893cee5f0976c030e33444149c40c5

File: e2e/bitrise.yml [hunk 1]
```diff
@@ -98,8 +98,9 @@ workflows:
 
   _start_emulator:
     steps:
-    - avd-manager:
+    - git::https://github.com/bitrise-steplib/steps-avd-manager@MOB-268-hanging-sdkmanager:
```
Comment (ofalvai) [inline e2e/bitrise.yml:101]: Should we merge that to master?

**Review by ofalvai — APPROVED** [review]

**Review by mateherber — CHANGES_REQUESTED** [review]

File: commands.go [hunk 2]
```diff
@@ -22,30 +26,39 @@ func (r defaultCmdRunner) RunCommandWithTimeout(name string, args []string) (str
...
+	log.TPrintf("Started: %s", strings.Join(cmd.Args, " "))
+
 	done := make(chan error)
 
 	go func() { done <- cmd.Wait() }()
+
 	select {
 	case err := <-done:
+		if err != nil {
+			log.TPrintf("Failed with output: %s, error: %s", strings.TrimSpace(output.String()), err)
```
[the remark this answers is not in the document]
Comment (godrei) [reply commands.go]: updated in https://github.com/bitrise-steplib/steps-wait-for-android-emulator/pull/25/commits/c6dc3149fe813c7ebfee62dd46fb447b98872fed

**Review by mateherber — DISMISSED** [review]

**Comment by DamienBitrise** [comment]
Any reason why this was never released?

**Comment by godrei** [comment]
@DamienBitrise more recent PR related to this was merged: https://github.com/bitrise-steplib/steps-wait-for-android-emulator/pull/28

**Pull request closed without merging**